clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -847,6 +864,13 @@ public void handleDisconnection(String destination) {
         }
 
         @Override
+        public void handleAuthenticationFailure(ApiException exception) {
```

Comment:
Hmm.. I wonder if we may as well use the more specific `AuthenticationFailedException` rather than `ApiException` everywhere.

Reply:
To remain consistent with the exception type used in `KafkaChannel.java` I replaced `ApiException` with `AuthenticationException` (which is a super-type of `AuthenticationFailedException`).